assignment5.txt

```diff
@@ -0,0 +1,20 @@
+1) To call the function, it would look like this:
+
+    /* Misc stuff above the main function, including functions, function
+    * declarations, header statements (like #define and #include), and stuff
+    */
+    int main(){
+        /* we need 2 ints and 1 float to pass to the function. */
+        int a = 6;
+        float b = 0.5;
+        int c = 2;
+        /* the function returns an integer value */
+        printf("%d\n", fxn(a, b, c));
+    }
+    /* Misc stuff after the main function */
+
+2) Looping is usually better, as it doesn't require the computer to store the previous
+iterations in memory. Recursion would be preferable for things like trees, where it
+needs to go down more than one path.
+
+3) no idea.
```

Comment:
In short, a compiler converts a program from a human-readable format into a machine-readable format. 
Lexing - break up the text of the program into "tokens" 
Parsing - convert the sequence of tokens into a parse tree, which is a data structure representing various language constructs: type declarations, variable declarations, function definitions, loops, conditionals, expressions, etc. 
Optimization - evaluate constant expressions, optimize away unused variables or unreachable code, unroll loops if possible, etc. 
Translate the parse tree into machine instructions (or JVM byte code)